tests/e2e/graph/introspect.rs

```diff
@@ -72,33 +75,56 @@ async fn e2e_test_rover_graph_introspect(
 #[traced_test]
 async fn e2e_test_rover_graph_introspect_watch(
     #[from(run_single_mutable_subgraph)]
-    #[future]
-    subgraph_details: (String, TempDir, String),
+    #[future(awt)]
+    subgraph: SingleMutableSubgraph,
     test_artifacts_directory: PathBuf,
+    introspection_log_line_prefix: &Regex,
 ) {
-    // Set up the command to output the original file
+    // Create an output file to hold the introspection responses
     let mut out_file = Builder::new()
         .suffix(".json")
         .tempfile()
         .expect("Could not create output file");
-    let (url, subgraph_dir, schema_name) = subgraph_details.await;
+    // Create the Rover command to run the introspection in `--watch` mode
     let mut cmd = Command::cargo_bin("rover").expect("Could not find necessary binary");
-    cmd.args([
-        "graph",
-        "introspect",
-        &url,
-        "--watch",
-        "--format",
-        "json",
-        "--output",
-        out_file.path().to_str().unwrap(),
-    ]);
-    let mut child = cmd.spawn().expect("Could not run command");
-    tokio::time::sleep(Duration::from_secs(1)).await;
-    // Store the result
+    let mut child = cmd
+        .args([
+            "graph",
+            "introspect",
+            &subgraph.subgraph_url,
+            "--watch",
+            "--format",
+            "json",
+            "--output",
+            out_file.path().to_str().unwrap(),
+        ])
+        .stderr(Stdio::piped())
+        .spawn()
+        .expect("Could not run rover command");
+    info!("Running rover introspection command...");
+
+    // Extract stderr from the child process and attach a reader to it so we can explore
+    // the lines of output
+    while child.stderr.is_none() {
```

Comment:
Is it worth adding a timeout here? Just in case this never returns `false`

Reply:
Yer that's a great idea, have implemented